test/integration/targets/setup_deb_repo/tasks/main.yml

```diff
@@ -0,0 +1,36 @@
+- block:
+  - name: Install needed packages
+    apt:
+      name: "{{ item }}"
+    with_items:
+      - dpkg-dev
+      - equivs
+      - libfile-fcntllock-perl  # to silence warning by equivs-build
+
+  - set_fact:
+      repodir: /tmp/repo/
+
+  - name: Create repo dir
+    file:
+      path: "{{ repodir }}"
+      state: directory
+      mode: 0755
+
+  - name: Create deb files
+    shell: "equivs-build {{ item }}"
+    args:
+      chdir: "{{ repodir }}"
+    with_fileglob:
+      - "files/package_specs/*"
+
+  - name: Create repo
+    shell: dpkg-scanpackages --multiversion . /dev/null | gzip -9c > Packages.gz
+    args:
+      chdir: "{{ repodir }}"
+
+  - name: Install the repo
+    apt_repository:
```

Comment:
This repo should be removed when the test ends (success or failure) along with the installed package(s).

Reply:
Thanks for the review, should be fixed. Also did that in the yum tests - https://github.com/ansible/ansible/pull/37770.